sample/legacy_with_buckets.go

```diff
@@ -0,0 +1,35 @@
+package sample
+
+import (
+	metrics "github.com/rcrowley/go-metrics"
+)
+
+type LegacyWithBuckets interface {
+	metrics.Sample
+	BucketsAndValues() (buckets []float64, values []int64)
+}
+
+type legacyWithBuckets struct {
+	metrics.Sample
+	bucketPartialSample
+}
+
+func NewLegacyWithBuckets(reservoirSize int, buckets ...float64) metrics.Sample {
+	return &legacyWithBuckets{
+		Sample:              NewLegacy(reservoirSize),
+		bucketPartialSample: mustNewBucketPartialSample(buckets...),
+	}
+}
+
+func (s *legacyWithBuckets) Update(val int64) {
```

Comment:
as agreed with @akaspin we go with `int64` intentionally for a backward compatibility with an existing code. And we do not add floats anywhere for now.